### ISSUE TITLE
Document.getSelectedDocument does not point to current document since 53.1

### ISSUE THREAD

**Issue by ryenbeatty** [BohemianCoding/SketchAPI#386]
Hi there,

I should clarify that this does not happen in all situations, rather in this specific one.

I have an action that fires on OpenDocument that bootstraps an instance of BrowserWindow (from sketch-module-webview). I then have a plugin command which accesses that instance by it's ID, and then shows it.

The simplified code looks like this:

```
const generateWindow = () => {
    const window = new BrowserWindow({
      identifier: 'my-identifier',
      show: false,
      ...
    });

    win.on('focus', () => {
      const document = Document.getSelectedDocument(); // always references the first opened document
    });
    return window;
}

export const onDocumentOpened = () => {
    generateWindow();
}

export default () => {
    const window = BrowserWindow.fromId('my-identifier');
    window.show();
}
```

The problem seems to be that any calls to Document.getSelectedDocument after the window is created, always point to the first document that was opened, so if any subsequent documents are opened then we are not able to select them.

Please let me know if you need any further clarification, but it seems to me that something related to the JS runtime inside Sketch has changed.

Thanks!
Ryen

**Comment by mathieudutour** [comment]
Hum in 53.1, the `Document.getSelectedDocument()` will return the document that is in the context because of this issue: https://github.com/BohemianCoding/SketchAPI/issues/253
But I realise that picking it from the context as the first choice is not ideal in the use case above (a long running plugin that access the selected document at different point of its life)

**Comment by ryenbeatty** [comment]
In this case, even if I replaced my Document.getSelectedDocument call with Document.fromNative(context.document), the reference would still be to the original (i.e. first) document that was opened.

I have worked around it by keeping a reference to the current document by id, and generating one window per document. This works fine as I can just lookup the current document by id, from Document.getDocuments.
